lib/blockchain.rb

```diff
@@ -34,7 +34,7 @@ def hash_transaction(data, time)
   end
 
   def create_block(block_class)
-    block = block_class.new(unverified_transactions, blocks.last.hash)
+    block = block_class.new(unverified_transactions.dup, blocks.last.hash)
```

Comment:
what is dup?

Reply:
It creates a copy of the object, rather than rely on the original object. The problem we were having is the 'transactions' for each block were referencing the 'unverified transactions' list. What dup does here is it copies the unverified transactions and passes them to the block.

Reply:
Get it!